indexer/packages/postgres/src/stores/transfer-table.ts

```diff
@@ -330,6 +331,126 @@ export async function findAllToOrFromSubaccountId(
   };
 }
 
+/**
+ * Finds all transfers to or from child subaccounts of a parent subaccount,
+ * excluding transfers between child subaccounts of the same parent.
+ *
+ * @param subaccountId - Array of all child subaccount IDs for the parent
+ * @param limit - Maximum number of results to return
+ * @param createdBeforeOrAtHeight - Filter transfers created at or before this height
+ * @param createdBeforeOrAt - Filter transfers created at or before this time
+ * @param page - Page number for pagination
+ *
+ * @returns Paginated list of transfers with same-parent transfers filtered out
+ */
+export async function findAllToOrFromParentSubaccount(
+  {
+    subaccountId,
+    limit,
+    createdBeforeOrAtHeight,
+    createdBeforeOrAt,
+    page,
+  }: ParentSubaccountTransferQueryConfig,
+  requiredFields: QueryableField[],
+  options: Options = DEFAULT_POSTGRES_OPTIONS,
+): Promise<PaginationFromDatabase<TransferFromDatabase>> {
+  verifyAllRequiredFields(
+    {
+      [QueryableField.LIMIT]: limit,
```

Comment:
Should we set an upper limit for this? and validate that limit is greater than 0